Description: Fix kfreebsd build skipping malloc_np.h include
Author: Jean Baptiste Favre <debiabn@jbfavre.org>
Origin: other
Bug: https://github.com/apache/trafficserver/pull/4327
Forwarded: https://github.com/apache/trafficserver/pull/4327
Applied-Upstream: https://github.com/apache/trafficserver/commit/b8d71fdec73d61b0bd5220f31ae4280a2c75fc49
Last-Update: 2018-10-04
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/tscore/ink_memory.cc
+++ b/src/tscore/ink_memory.cc
@@ -27,7 +27,7 @@
 #include "tscore/Diags.h"
 #include "tscore/ink_atomic.h"
 
-#if defined(freebsd)
+#if !defined(kfreebsd) && defined(freebsd)
 #include <malloc_np.h> // for malloc_usable_size
 #endif
 
--- a/proxy/Plugin.cc
+++ b/proxy/Plugin.cc
@@ -126,7 +126,7 @@ plugin_load(int argc, char *argv[], bool
       return false; // this line won't get called since Fatal brings down ATS
     }
 
-#if defined(freebsd) || defined(darwin)
+#if (!defined(kfreebsd) && defined(freebsd)) || defined(darwin)
     optreset = 1;
 #endif
 #if defined(__GLIBC__)
--- a/proxy/http/remap/RemapConfig.cc
+++ b/proxy/http/remap/RemapConfig.cc
@@ -897,7 +897,7 @@ remap_load_plugin(const char **argv, int
   void *ih         = nullptr;
   TSReturnCode res = TS_SUCCESS;
   if (pi->fp_tsremap_new_instance) {
-#if defined(freebsd) || defined(darwin)
+#if (!defined(kfreebsd) && defined(freebsd)) || defined(darwin)
     optreset = 1;
 #endif
 #if defined(__GLIBC__)
--- a/plugins/tcpinfo/tcpinfo.cc
+++ b/plugins/tcpinfo/tcpinfo.cc
@@ -136,7 +136,7 @@ log_tcp_info(Config *config, const char
   TSReturnCode ret;
 
   if (config->log_level == 2) {
-#if !defined(freebsd) || defined(__GLIBC__)
+#if !defined(kfreebsd) && (!defined(freebsd) || defined(__GLIBC__))
     ret = TSTextLogObjectWrite(config->log, "%s %s %s %u %u %u %u %u %u %u %u %u %u %u %u %u", event_name, client_str, server_str,
                                info.tcpi_rtt, info.tcpi_rttvar, info.tcpi_last_data_sent, info.tcpi_last_data_recv,
                                info.tcpi_snd_cwnd, info.tcpi_snd_ssthresh, info.tcpi_rcv_ssthresh, info.tcpi_unacked,
